plugins/modules/proxmox_vm_info.py

```diff
@@ -0,0 +1,221 @@
+author: 'Sergei Antipov (@UnderGreen) <greendayonfire at gmail dot com>'
+options:
+  node:
+    description:
+      - Node where to get virtual machines info.
+    required: true
+    type: str
+  type:
+    description:
+      - Restrict results to a specific virtual machine(s) type.
+    type: str
+    choices:
+      - all
+      - qemu
+      - lxc
+    default: all
+  vmid:
+    description:
+      - Restrict results to a specific virtual machine by using its ID.
+    type: int
+  name:
+    description:
+      - Restrict results to a specific virtual machine by using its name.
+      - If multiple virtual machines have the same name then vmid must be used instead.
+    type: str
+extends_documentation_fragment:
+    - community.general.proxmox.documentation
+    - community.general.attributes
+    - community.general.attributes.info_module
+"""
+
+EXAMPLES = """
+- name: List all existing virtual machines on node
+  community.general.proxmox_vm_info:
+    api_host: proxmoxhost
+    api_user: root@pam
+    api_token_id: '{{ token_id | default(omit) }}'
+    api_token_secret: '{{ token_secret | default(omit) }}'
+    node: node01
+
+- name: List all QEMU virtual machines on node
+  community.general.proxmox_vm_info:
+    api_host: proxmoxhost
+    api_user: root@pam
+    api_password: '{{ password | default(omit) }}'
+    node: node01
+    type: qemu
+
+- name: Retrieve information about specific VM by ID
+  community.general.proxmox_vm_info:
+    api_host: proxmoxhost
+    api_user: root@pam
+    api_password: '{{ password | default(omit) }}'
+    node: node01
+    type: qemu
+    vmid: 101
+
+- name: Retrieve information about specific VM by name
+  community.general.proxmox_vm_info:
+    api_host: proxmoxhost
+    api_user: root@pam
+    api_password: '{{ password | default(omit) }}'
+    node: node01
+    type: lxc
+    name: lxc05.home.arpa
+"""
+
+RETURN = """
+proxmox_vms:
+    description: List of virtual machines.
+    returned: on success
+    type: list
+    elements: dict
+    sample:
+      [
+        {
+          "cpu": 0.258944410905281,
+          "cpus": 1,
+          "disk": 0,
+          "diskread": 0,
+          "diskwrite": 0,
+          "maxdisk": 34359738368,
+          "maxmem": 4294967296,
+          "mem": 35158379,
+          "name": "pxe.home.arpa",
+          "netin": 99715803,
+          "netout": 14237835,
+          "pid": 1947197,
+          "status": "running",
+          "type": "qemu",
+          "uptime": 135530,
+          "vmid": 100
+        },
+        {
+          "cpu": 0,
+          "cpus": 1,
+          "disk": 0,
+          "diskread": 0,
+          "diskwrite": 0,
+          "maxdisk": 0,
+          "maxmem": 536870912,
+          "mem": 0,
+          "name": "test1",
+          "netin": 0,
+          "netout": 0,
+          "status": "stopped",
+          "type": "qemu",
+          "uptime": 0,
+          "vmid": 101
+        }
+      ]
+msg:
```

Comment:
`msg` is a common return value and doesn't need to be documented (if it doesn't have a special meaning).